uoftscrapers/scrapers/events/__init__.py

```diff
@@ -0,0 +1,115 @@
+from collections import OrderedDict
+import urllib.parse as urlparse
+from urllib.parse import urlencode
+import re
+
+class Events:
+    """A scraper for Events at the University of Toronto."""
+    host = 'https://www.events.utoronto.ca/'
+
+    @staticmethod
+    def scrape(location='.'):
+        Scraper.logger.info('Events initialized.')
+        Scraper.ensure_location(location)
+       
+        for event_link in Events.get_events_links():
+            doc = Events.get_event_doc(event_link)
+            Scraper.save_json(doc, location, doc['id'])
+
+        Scraper.logger.info('Events completed.')
+
+    @staticmethod
+    def get_events_links():
+        page_index_url = Events.host + 'index.php'
+        url_parts = list(urlparse.urlparse(page_index_url))
+        events_links = []
+        paging_index = 1
+        events_count = 10 
+        while(events_count == 10):
+            params = {
+                    'p': paging_index
+                }
+            url_parts[4] = urlencode(params)
+            paging_index += 1
+            html = Scraper.get(urlparse.urlunparse(url_parts))
+            soup = BeautifulSoup(html, 'html.parser')
+            events_dom_arr = soup.select('#results')[0].find_all('li')
+            events_count = len(events_dom_arr)
+            events_links += list(map(lambda e: e.a['href'], events_dom_arr))
+        return(events_links)                
+
+    @staticmethod
+    def get_event_doc(url_tail):
+        event_url = Events.host + url_tail
+        html = Scraper.get(event_url)
+        url_parts = list(urlparse.urlparse(event_url))
+        query = dict(urlparse.parse_qsl(url_parts[4]))
+        soup = BeautifulSoup(html, 'html.parser')
+
+        event_id = query['eventid']
+        event_title = soup.select('.eventTitle')[0].text.strip()
+        raw_time = soup.select('.date')[0].text.split(',')
+        
+        date_arr = raw_time[0].split(' - ')
+        time_arr = re.split(' - | ', raw_time[1].strip())
+        
+        # Some of the strings are misformed and gives an extra empty space
+        time_arr = list(filter(None, time_arr))
+        event_start_date = datetime.strptime(date_arr[0],
+            '%b %d').replace(year=date.today().year).date().isoformat()
+        event_end_date = datetime.strptime(date_arr[-1], 
+            '%b %d').replace(year=date.today().year).date().isoformat()
+
+        # Note: Some events span across several days e.g. 8350, thus specifying dates makes no sense
+        event_meridiem = time_arr[2]
+        event_start_time = time_arr[0] + ' ' + event_meridiem
+        event_end_time = time_arr[1] + ' ' + event_meridiem
+        
+        evt_bar = soup.select('#evt_bar')[0]
+        event_url = evt_bar.select('dd')[1].a['href']
+        event_price = evt_bar.select('dl')[1].dd.text
+
+        event_campus = 'Off Campus'
+        if evt_bar.select('dd')[0].b != None:
+            event_campus = evt_bar.select('dd')[0].b.text
```

Comment:
`event_campus` should take on a value of `UTSG`, `UTM`, or `UTSC` depending on the campus (instead of `St. George`, etc.) to follow convention.

In the case of no campus / off campus, we can just leave `event_campus = ''`